docs/Dialects/Sim/RationaleSim.md

```diff
@@ -0,0 +1,22 @@
+# Simulation (Sim) Dialect Rationale
+
+This document describes various design points of the `sim` dialect, why it is
+the way it is, and current status. This follows in the spirit of other [MLIR
+Rationale docs](https://mlir.llvm.org/docs/Rationale/).
+
+## Introduction
+
+The `sim` dialect provides a high-level representation for simulator-specific
+operations. The purpose of the dialect is to provide a high-level representation
+for constructs which interact with simulators (Verilator, VCS, Arc, ...) that
+are easy to analyze and transform in the compiler.
+
+## Operations
+
+### Plusargs
+
+The `sim.plusarg_test` and `sim.plusarg_value` operations are wrappers around
+the SystemVerilog built-ins which access command-line arguments.
```

Comment:
This part is slightly anti-motivating... Usually an op can justify itself without having to say that it is wrapper around several ops in another dialect.

Reply:
We discussed offline - we'll keep these ops as they are due to constraints from frontend intrinsics, but we need to change the system.